### PR TITLE
[FIX] web: fix incorrect rounding of scrollHeight

### PR DESCRIPTION
**Pull request by feelwhy**
Description of the issue/feature this PR addresses:
Building block can't be selected in html fields (web_editor/static/static/src/js/field_html.js) if window scale is more than 100% (e.g. 125%)
The error occurs (for example, in mass_mailing):
"""
Uncaught TypeError: Cannot read property 'contains' of undefined
at Class._onSmoothDragStart (web.assets_common.js:5048)
at HTMLDivElement.start (web.assets_common.js:5028)
at $..._trigger (web.assets_common.js:2090)
at $..._trigger (web.assets_common.js:2229)
at $..._trigger (web.assets_common.js:2060)
at $..._mouseStart (web.assets_common.js:2153)
at $..._mouseStart (web.assets_common.js:2060)
at $..._mouseMove (web.assets_common.js:2137)
at $..._mouseMove (web.assets_common.js:2060)
at HTMLDocument._mouseMoveDelegate (web.assets_common.js:2134)
"""

Current behavior before PR:
When searching for scrolling element, the rounding of the scrollHeight results in incorrect check between height() and actual scrolling element height (e.g. 598 < 598.4).
As a result, web_editor field_html widget works improperly when the scale of window is more than 100%.

Desired behavior after PR is merged:
Scrolling element height is checked without rounding.
HTML fields (including mass_mailing html) work properly

Links:

#63306 - issue registered for mass_mailing blocks its usage
https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollHeight - documentation explanation why getBoundingClientRect().height should be used instead of scrollHeight (the blue warning regarding rounding)

Also: [CLA] Individual signature for Alexander Filippov (feelwhy)

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by feelwhy** [comment]
Hello!
Any news regarding? Is it possible to somehow consider the PR?
It is a single-line change commit, which is reasoned by a direct reference from MDN web documentation. A lot of our clients have to work in Odoo with zoomed screen because of that.
Thanks!

File: addons/web/static/src/js/libs/jquery.js [hunk 1]
```diff
@@ -171,7 +171,7 @@ $.fn.extend({
             // Search for a body child which is at least as tall as the body
             // and which has the ability to scroll if enough content in it. If
             // found, suppose this is the top scrolling element.
-            if (el.scrollHeight < bodyHeight) {
+            if (el.getBoundingClientRect().height < bodyHeight) {
```
Comment (qsm-odoo) [inline addons/web/static/src/js/libs/jquery.js:174]: For the sake of properly comparing float numbers, maybe we could go with this?
```javascript
if (bodyHeight - el.getBoundingClientRect().height > 1) { 
```
And in that case we could still use `el.scrollHeight` I guess to change as less as possible in stable versions. In that case, you can remove the MDN doc link from your commit message.

**Review by qsm-odoo — CHANGES_REQUESTED** [review]
Hello @feelwhy 

Sorry for the delay checking the related issue, we try to have moments when we check issues on GitHub but unfortunately we don't have the time to do it as much as we'd like. However, if needed you can try pinging me on issues you think are related to me, that would be more efficient in general than sending me long emails I just might skip by mistake :flushed:

I added a suggestion on the PR. Could you also adapt your commit message to limit to number of characters per line to 72 (if you prefer 80, it's fine too)?

Thank you!

**Comment by feelwhy** [comment]
Hello! 

Surely, your suggestion is totally fine for us (actually, it was the one of the first things we offered in the https://github.com/odoo/odoo/issues/63306).

For us it also totally fine if you introduced that change by yourself not within a PR (since PR in that case does have other details). 

Otherwise, let me know and I would amend commit for the message-length requirement and with your suggestion.

Thank you a lot!

**Comment by qsm-odoo** [comment]
As you prefer, if you want your CLA commit merged, it may be easier that I let you amend the commit with the suggestions and merge both commits at the same time. I can also make the commit and merge it only by myself, as you wish, just tell me :+1:

**Comment by feelwhy** [comment]
It would be great! I made commits through GitHub (so need to clone everything and amend locally). It would be faster if you did this. Thank you again a lot for your help!

**Comment by qsm-odoo** [comment]
Merging with https://github.com/odoo/odoo/pull/67961

**Pull request closed without merging**